### ISSUE TITLE
Overflow/underflow issues

### ISSUE THREAD

**Issue by andreped** [andreped/GradientAccumulator#35]
When training deep neural networks in general, overflow/underflow can occur.

When we are accumulating gradients, it is even more likely that this can occur, and can greater impact on training, as the overflow/underflow can be more severe.

This is also relevant for mixed precision, as can be seen discussed in the [TensorFlow docs](https://www.tensorflow.org/guide/mixed_precision#underflow_and_overflow).

For mixed precision, TensorFlow uses dynamic loss scaling to reduce the chance of underflow to happen during backpropagation.

As we are also scaling the loss, and thus the gradients, during gradient accumulation, we likely need to do this scaling more carefully, such that we don't run into underflow. There is a discussion about this in [this thread](https://github.com/keras-team/tf-keras/issues/301). We should try to incorporate some of those suggestions to further improve the robustness of our method.

**Comment by andreped** [comment]
I believe our implementations have matured enough to take this into account. There has likely been proposed tricks in recent publications on how to achieve better approximations, but before we have a clear issue with one of our implementations or proposed fix, I believe this issue can be closed, as it serves no value having it open.

Feel free to reopen if you find an issue related to this or propose a fix.
